From 030c6cfec2977b3f64d198d88b97246a9209c278 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Thu, 7 Dec 2017 18:16:49 -0600
Subject: [PATCH] fix makefile

Remove CC and AR definitions from the Makefile, use from the environment
unset prefix

Rebased for 1.0.0 (context shifted, fix unchanged). Also gates LDL and
VERSION_OPT on Darwin: 1.0.0 added -ldl and -Wl,-soname unconditionally,
but macOS has no libdl and its linker doesn't support -soname.
---
 libraries/liblmdb/Makefile | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/libraries/liblmdb/Makefile b/libraries/liblmdb/Makefile
index f254511..fc465f2 100644
--- a/libraries/liblmdb/Makefile
+++ b/libraries/liblmdb/Makefile
@@ -18,8 +18,8 @@
 # There may be other macros in mdb.c of interest. You should
 # read mdb.c before changing any of them.
 #
-CC	= gcc
-AR	= ar
+#CC	= gcc
+#AR	= ar
 W	= -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized
 THREADS = -pthread
 OPT = -O2 -g
@@ -28,7 +28,12 @@
 LDLIBS	= 
 SOLIBS	= 
 SOEXT	= .so
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Darwin)
+LDL		=
+else
 LDL		= -ldl
+endif
 
 LIBVER	= 1
 ABIVER	= 0
@@ -36,7 +41,7 @@
 LMDB_VERSION	= 1.0.0
 SOFULL	= $(SOEXT).$(VEREXT)
 
-prefix	= /usr/local
+prefix	=
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
 libdir = $(exec_prefix)/lib
@@ -54,7 +59,11 @@
 PROGS	= $(IPROGS) mtest mtest2 mtest3 mtest4 mtest5
 RPROGS	= mtest_remap mtest_enc mtest_enc2
 SOVER	= liblmdb$(SOEXT).$(LIBVER)
+ifeq ($(UNAME_S),Darwin)
+VERSION_OPT	=
+else
 VERSION_OPT	= -Wl,-soname,$(SOVER)
+endif
 # For MacOSX:
 #VERSION_OPT	= -Wl,-current_version,$(VEREXT)
 
--
2.11.1
